lib/logitech_receiver/base.py

```diff
@@ -189,6 +198,23 @@ def read(handle, timeout=DEFAULT_TIMEOUT):
 		return reply[1:]
 
 
+# sanity checks on  message report id and size
+def check_message(data) :
+	assert isinstance(data, bytes), (repr(data), type(data))
+	report_id = ord(data[:1])
+	if report_id in report_lengths:  # is this an HID++ or DJ message?
+		if report_lengths.get(report_id) == len(data):
```

Comment:
Join the two ifs.

Reply:
If the report id is in report lengths there are two cases:
correct size - return true
incorrect size - warn and return false
Combining into one if would combine incorrect size for known report ids (10,11,20,21) and unknown report ids.   So the logging would have to be the same for both.  
I think that malformed known reports need logging and warning is probably the right level.  Unknown reports don't need logging, I think.

Reply:
Ah, yes, sorry :)